iothub/device/src/Pipeline/TransportHandler.cs

```diff
@@ -25,15 +25,30 @@ public override Task WaitForTransportClosedAsync()
 
         protected override void Dispose(bool disposing)
         {
-            if (_disposed)
+            try
             {
-                return;
-            }
+                if (Logging.IsEnabled)
+                {
+                    Logging.Enter(this, $"{nameof(DefaultDelegatingHandler)}.Disposed={_disposed}; disposing={disposing}", $"{nameof(TransportHandler)}.{nameof(Dispose)}");
```

Comment:
Why `DefaultDelegatingHandler`?

Reply:
The `_disposed` parameter referenced throughout our delegating handler pipeline is actually the protected variable defined in `DefaultDelegatingHandler`: https://github.com/Azure/azure-iot-sdk-csharp/blob/929a6d36ee8dbf9f66f0a7829f8c4c0c489a9b29/iothub/device/src/Pipeline/DefaultDelegatingHandler.cs#L15

This, along with the way `base.Dispose()` and `inherited.Dispose()` are mixed in our pipeline disposal flow is something I'm looking to refactor.